luigi_pipeline/lib/model/base_mt_schema.py

```diff
@@ -0,0 +1,105 @@
+from abc import abstractmethod
+from inspect import getmembers, isfunction
+from functools import wraps
+
+
+def mt_annotation(annotation=None, fn_require=None):
+    """
+    Function decorator for methods in a subclass of BaseMTSchema.
+    Allows the function to be treated like an mt_annotation with annotation name and value.
+
+        @mt_annotation()
+        def a(self):
+            return 'a_val'
+
+        @mt_annotation(annotation='b', fn_require=a)
+        def b_1(self):
+            return 'b_val'
+
+    Will generate a mt with rows of {a: 'a_val', 'b': 'b_val'} if the function is called.
+    TODO: Consider changing fn_require to be a list of requirements.
+
+    :param annotation: name in the final MT. If not provided, uses the function name.
+    :param fn_require: method name strings in class that are dependencies.
+    :return:
+    """
+    def mt_prop_wrapper(func):
+        annotation_name = annotation or func.__name__
+
+        @wraps(func)
+        def wrapper(self, *args, **kwargs):
+            # Called already.
+            if wrapper.mt_prop_meta['annotated'] > 0:
+                return self
+            if fn_require:
+                getattr(self, fn_require.__name__)()
```

Comment:
This seems unintuitive given that it looks like an arbitrary function can be passed as a requirement but it must actually be a method of the same schema class.

Reply:
Yeah, I thought to add a check to make sure it's a method of the class, but figured both would fail with essentially the same error message. I'll add an informative exception to be explicit.